solc/CommandLineParser.cpp

```diff
@@ -687,7 +690,8 @@ General Information)").c_str(),
 		)
 		(
 			g_strImportEvmAssemblerJson.c_str(),
-			"Import EVM assembly from JSON. Assumes input is in the format used by --asm-json."
+			("Import EVM assembly from JSON. Assumes input is in the format used by --asm-json. Providing "
+			"--" + g_strOptimize + " will run the optimizer again and may change results.").c_str()
```

Comment:
How about this?

```suggestion
			("Import EVM assembly in JSON format produced by --asm-json. "
			"WARNING: --asm-json output is already optimized according to settings stored in metadata. "
			"Using --" + g_strOptimize + " in this mode is allowed, but not necessary under normal circumstances. "
			"It forces the optimizer to run again and produces bytecode that's not reproducible from metadata.").c_str()
```

Reply:
We also need a similar warning on the `optimizer` dict in Standard JSON docs. Using a non-default value for anything there will pretty much have the same effect.

Reply:
Yeah that makes sense. Added a warning.